### DISCUSSION TITLE
I want to be able to debug an app and step into the maui repo code.

### DISCUSSION THREAD

**Discussion by davefxy** [dotnet/maui#10646] [Q&A]
I have tried reading and following the steps in the Develoment.md document.
1. forked the dotnet maui repo, net 7 branch.
2. exec these requests from command :
  dotnet tool restore
   dotnet cake --target=VS

What would be my next step if I want to run the control samples in debug mode stepping through the apps and into maui repo system code? i.e stepping through the map handlers.

**Comment by philipag** [comment]
It would be useful if even without compiling MAUI one could step into iOS/Android framework code like one can with .Net under windows.  This would make tracking down MAUI bugs much easier.  As-is one has to compile the entire framework.

Last time I tried following https://github.com/dotnet/maui/blob/main/.github/DEVELOPMENT.md#testing-branch-against-your-project I ran into lots of problems and gave up.   Just had a look again and the instructions simply are not clear.  How do I create a project-local MAUI pack for the latest VS preview in an existing project?  How do I switch back and forth between the released preview pack that comes with VS preview and my locally built pack?

**Comment by philipag** [comment]
I tried following DEVELOPMENT.MD again and cannot get things to work.

First of all, the document appears to describe things in reverse order which is very confusing.  It first talks about "dotnet tool restore" etc and AFTER that it talks about cloning the MAUI repository.  Because of that I was under the impression that  "dotnet tool restore" is to be run on my project using a VS powershell, but that is wrong?

So then I followed the document from back to front and first cloned "release/7.0.1xx-sr1.1", opened a VS powershell in that folder and ran "dotnet tool restore" and "dotnet cake --target=VS".  That worked  without errors and automatically opened VS with the MAUI solution.  Then I pressed F5 in VS to debug the default startup project "Maui.Control.Sample" but there were lots of compile time errors for various projects.  I tried this a few months ago and it also failed (back then there were problems with certificates which thankfully is no longer the case).

Can anyone actually follow DEVELOPMENT.MD to get a working environment?  It really seems that this document and process need to be reworked a bit to make local MAUI builds more accessible to people who might want to contribute code.

**Reply by hugebug4ever** [reply]
The Maui.Control.Sample work for me, with `git checkout tags/7.0.52`. 
It just worked as the document described.
I am struggling to figure out how the specify the dotnet sdk environment. Seems it points to 7.0.100-rtm.22519.39 instead of my global environment (7.0.102)
